src/esp/bindings/GfxReplayBindings.cpp

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#include "esp/bindings/bindings.h"
+
+#include <Magnum/PythonBindings.h>
+#include <Magnum/SceneGraph/PythonBindings.h>
+
+#include "esp/gfx/replay/Player.h"
+#include "esp/gfx/replay/ReplayManager.h"
+
+namespace py = pybind11;
+using py::literals::operator""_a;
+
+namespace esp {
+namespace gfx {
+namespace replay {
+
+void initGfxReplayBindings(py::module& m) {
+  py::class_<Player, Player::ptr>(m, "Player")
+      .def("get_num_keyframes", &Player::getNumKeyframes,
+           R"(Get the currently-set keyframe, or -1 if no keyframe is set.)")
+
+      .def(
+          "set_keyframe_index", &Player::setKeyframeIndex,
+          R"(Set a keyframe by index, or pass -1 to clear the currently-set keyframe.)")
+
+      .def("get_keyframe_index", &Player::getKeyframeIndex,
+           R"(Get the number of keyframes read from file.)")
+
+      .def(
+          "get_user_transform",
+          [](Player& self, const std::string& name) {
+            Magnum::Vector3 translation;
+            Magnum::Quaternion rotation;
+            bool found = self.getUserTransform(name, &translation, &rotation);
+            return found ? py::make_tuple(translation, rotation)
+                         : py::make_tuple(nullptr, nullptr);
+          },
+          R"(todo)");
+
+  py::class_<ReplayManager, ReplayManager::ptr>(m, "ReplayManager")
+      .def(
+          "save_keyframe",
+          [](ReplayManager& self) {
+            if (!self.getRecorder()) {
+              LOG(ERROR) << "save_keyframe: not enabled. See "
+                            "SimulatorConfiguration::enableGfxReplaySave.";
+              return;
+            }
```

Comment:
shouldn't all these `LOG(ERROR)` be rather exceptions instead of printing an error and silently doing nothing? i feel that's more pythonic

Reply:
Sounds good, I'll investigate.

Reply:
Pybind should have most of the usual Python exceptions exposed, so it should be as easy as `throw`ing one of them: https://pybind11.readthedocs.io/en/stable/advanced/exceptions.html

Reply:
I throw std::runtime_error, which pybind turns into a python RuntimeError.